src/core/credGraph.js

```diff
@@ -70,9 +79,28 @@ export class CredGraph {
     }
   }
 
-  *scoringNodes(): Iterator<Node> {
-    for (const addr of this._mpg.scoringAddresses()) {
-      yield NullUtil.get(this.node(addr));
+  *participants(): Iterator<Participant> {
+    for (const {address, description} of this._mpg.participants()) {
+      const epochs = this._mpg.epochBoundaries().map((epochStart) => ({
+        type: "USER_EPOCH",
+        owner: address,
+        epochStart,
+      }));
+      let totalCred = 0;
+      const credPerEpoch = epochs.map((e) => {
+        const payoutEdgeAddress = payoutAddressForEpoch(e);
+        const payoutMarkovEdgeAddress = MarkovEdgeAddress.fromParts([
+          "F",
```

Comment:
Prior to this patch, the only occurrence of the `"B"`/`"F"` constants
was in the `rawEdgeAddress` function. I’m wary about duplicating it
here. What do you think about pulling out something like

```javascript
function edgeToMarkovEdge(
  edgeAddress: EdgeAddressT,
  direction: "B" /* Backward */ | "F" /* Forward */
): MarkovEdgeAddressT {
  return MarkovEdgeAddress.fromParts([
    direction,
    ...EdgeAddress.toParts(edge.address),
  ]);
}

function rawEdgeAddress(edge: MarkovEdge): MarkovEdgeAddressT {
  return edgeToMarkovEdge(edge.reversed ? "B" : "F", edge.address);
}
```

and then calling `edgeToMarkovEdge` directly here?

**EDIT:** Just noticed this is in `CredGraph`. In that case, I double
down: the `B`/`F` encoding is 100% an implementation detail of the MPG
module. OK to export this putative `edgeToMarkovEdge` function.